lib/saml_idp/request.rb

```diff
@@ -31,48 +31,129 @@ def initialize(raw_xml = "")
       self.raw_xml = raw_xml
     end
 
+    def logout_request?
+      logout_request.nil? ? false : true
+    end
+
+    def authn_request?
+      authn_request.nil? ? false : true
+    end
+
     def request_id
-      authn_request["ID"]
+      if authn_request?
+        authn_request["ID"]
+      elsif logout_request?
+        logout_request["ID"]
```

Comment:
Because these requests are somewhat similar, thoughts on having a single `request` location where that returns the proper xpath? They seem mutually exclusive as well. Might not need to expose the new boolean methods in that case?